stock_move_backdating/models/stock_inventory.py

```diff
@@ -0,0 +1,36 @@
+#  Copyright 2023 Simone Rubino - TAKOBI
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+from .stock_move_line import check_date
+
+
+class StockInventory(models.Model):
+    _inherit = "stock.inventory"
+
+    date_backdating = fields.Datetime(
+        string="Actual Inventory Date",
+    )
+
+    @api.onchange(
+        "date_backdating",
+    )
+    def onchange_date_backdating(self):
+        self.ensure_one()
+        check_date(self.date_backdating)
+
+    def post_inventory(self):
+        no_backdate_inventories = self.env["stock.inventory"].browse()
```

Comment:
That empty `browse` is not necessary, you get the same result by referencing the model

```suggestion
        no_backdate_inventories = self.env["stock.inventory"]
```